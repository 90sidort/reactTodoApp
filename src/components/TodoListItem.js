import React from "react";
import { Link } from "react-router-dom";

const TodoListItem = ({
  id,
  title,
  completed,
  details,
  priority,
  createdAt,
}) => {
  const colour = completed ? "#ffcccb" : "#9ACD32";
  return (
    <div style={{ background: colour }}>
      <Link to={`edit/${id}`}>
        <h3>Description:{title}</h3>
      </Link>
      <p>Details:{details}</p>
      <p>Priority: {priority}</p>
      <p>Creation Time:{createdAt}</p>
    </div>
  );
};

export default TodoListItem;
