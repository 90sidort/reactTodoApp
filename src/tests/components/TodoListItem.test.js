import TodoListItem from '../../components/TodoListItem'
import React from 'react'
import { shallow } from 'enzyme'
import mockTodos from '../fixtures/todosFixture'

test('Should correctly render TodoListItemComponent', () => {
    const wrapper = shallow(<TodoListItem {...mockTodos[0]}/>)
    expect(wrapper).toMatchSnapshot()
})

