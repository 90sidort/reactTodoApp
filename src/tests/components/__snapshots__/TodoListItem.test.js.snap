// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should correctly render TodoListItemComponent 1`] = `
<div
  style={
    Object {
      "background": "#ffcccb",
    }
  }
>
  <Link
    replace={false}
    to="edit/1"
  >
    <h3>
      Description:
      First todo
    </h3>
  </Link>
  <p>
    Details:
    These are first details
  </p>
  <p>
    Priority: 
    0
  </p>
  <p>
    Creation Time:
    259200000
  </p>
</div>
`;
